codebuild/spec/buildspec_unit_coverage.txt

```diff
@@ -0,0 +1,43 @@
+---
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+# this file except in compliance with the License. A copy of the License is
+# located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+version: 0.2
+
+env:
+  parameter-store:
+    CODECOV_TOKEN : codecov-upload-token
+  variables:
+    # CODEBUILD_ is a reserved namespace.
+    CB_BIN_DIR: "./codebuild/bin"
+    CC: "/usr/bin/clang"
+    CXX: "/usr/bin/clang++"
+
+phases:
+  pre_build:
+    commands:
+      - |
+        if [ -d "third-party-src" ]; then
+          cd third-party-src;
+          ln -s /usr/local $CODEBUILD_SRC_DIR/third-party-src/test-deps;
+        fi
+      - ln -s /usr/local $CODEBUILD_SRC_DIR/test-deps
+  build:
+    on-failure: ABORT
+    commands:
+      - $CB_BIN_DIR/coverage_build.sh
```

Comment:
While I love the divergence from our giant build script, I do have to ask: how are you imagining that this will scale to handle other libcryptos?

And what libcrypto does this currently use?

Reply:
I'll check with Doug on the libcrypto thing. I was previously under assumption that this would be passed in from the buildspec that we add to codebuild
```
S2N_LIBCRYPTO: openssl-1.1.1
```
But I think it might just be pulling in the default libcrypto from the ubuntu image.

Reply:
Made some change in the below commit, confirmed that we are now running on openssl-1.1.1, and made the build script a bit more general.

Reply:
Is the default openssl in the ubuntu image is also openssl-1.1.1?

Maybe try changing it to awslc to confirm it works? We might want to build with awslc anyway, since we want that to be our default.

Reply:
Uncertain about the default openssl for ubuntu image, but we can check the path that it found the libcrypto at
```
-- Found crypto: /codebuild/output/src400490645/src/github.com/jmayclin/s2n-tls/test-deps/openssl-1.1.1/lib/libcrypto.so  
-- LibCrypto Include Dir: /codebuild/output/src400490645/src/github.com/jmayclin/s2n-tls/test-deps/openssl-1.1.1/include
-- LibCrypto Shared Lib:  /codebuild/output/src400490645/src/github.com/jmayclin/s2n-tls/test-deps/openssl-1.1.1/lib/libcrypto.so
-- LibCrypto Static Lib:  /codebuild/output/src400490645/src/github.com/jmayclin/s2n-tls/test-deps/openssl-1.1.1/lib/libcrypto.a
-- Using libcrypto from the cmake path
```
which confirms that it is pulling from our test-dependencies section, which is what we wanted.

Also submitted a build using an AWS-LC buildspec 
```
    - identifier: s2nUnitCoverage
      buildspec: codebuild/spec/buildspec_unit_coverage.yml
      env:
        privileged-mode: true
        compute-type: BUILD_GENERAL1_LARGE
        image: 024603541914.dkr.ecr.us-west-2.amazonaws.com/docker:ubuntu22codebuild
        variables:
          S2N_LIBCRYPTO: awslc
```
Confirmed the following output
```
-- clone() support detected
-- FOUND AWS-LC CRYPTO cmake config - shared
-- Using libcrypto from the cmake path 
```